routes/models/proposal.js

```diff
@@ -0,0 +1,86 @@
+var md5= require('md5') ;
+var DB = require('../config.js') ;
+var random = require('../../libs/random.js') ;
+var Boom =  require('boom');
+
+module.exports.add = function(req,res) {
+    var proposal = req.payload ;
+    proposal.created =  new Date();
+    proposal.is_active = 1;
+    DB.conn.queryAsync("INSERT INTO "+DB.proposals +" SET ?" , proposal ).then(function(result)
+    {
+        res(result);
+    });
+}
+
+module.exports.proposals = function(req,res) {
+
+    if(!req.query.page)
+    {
+        res(Boom.badRequest('Please provide the page number'));
+    }
+
+    if(req.query.page <= 0)
+    {
+        res(Boom.badRequest('Invalid page number.'));
+    }
+
+    var prStatus = req.params.status ;
+    var perpage =  20;
+    var page = req.query.page ;
+    var search = false   ;
+    if(req.query.s) {
+      search = req.query.s ;
+    }
+
+    //if(req.query.perpage){ perpage = req.query.perpage; }
+    if(req.query.page){page = req.query.page ; }
+    var skip = perpage*(page-1) ;
+    var qry = "SELECT Proposals.* , User.id , User.first_name , Client.id,  Client.name FROM "+DB.proposals +" As Proposals INNER JOIN "+DB.users +" As User ON Proposals.assigned_to = User.id" + " INNER JOIN "+DB.clients + " As Client ON Proposals.client_id = Client.id where Proposals.is_active = "+prStatus ;    
+
+    if(search) {
+        qry += " AND (Proposals.title like '%"+search+"%'";
+        qry += " OR User.first_name like '%"+search+"%'";
+        qry += " OR User.last_name like '%"+search+"%'";
+        qry += " OR Client.name like '%"+search+"%') ";
+    }
```

Comment:
Added option for free search text based on Proposal title , client name , assigned User first name or last name